geomesa-feature/src/main/scala/org/locationtech/geomesa/feature/kryo/KryoFeatureSerializer.scala

```diff
@@ -113,6 +109,14 @@ object KryoFeatureSerializer {
 
   def apply(sft: SimpleFeatureType, decodeAs: SimpleFeatureType): KryoFeatureSerializer =
     if (sft.eq(decodeAs)) apply(sft) else  apply(new TransformingSimpleFeatureSerializer(sft, decodeAs))
+
+  def setupKryo(kryo: Kryo, serializer: Serializer[SimpleFeature]): Unit = {
+    kryo.setReferences(false)
+    kryo.register(classOf[ScalaSimpleFeature], serializer,  kryo.getNextRegistrationId)
+    kryo.register(classOf[KryoFeatureId], new FeatureIdSerializer(),  kryo.getNextRegistrationId)
+    kryo.register(classOf[SimpleFeature], serializer,  kryo.getNextRegistrationId)
+    kryo.register(classOf[AvroSimpleFeature], serializer,  kryo.getNextRegistrationId)
```

Comment:
So this class can de/serialize AvroSimpleFeatures, but can't deserialize an AvroSimpleFeature that was serialized in Avro...
@dmcwhorter did you try the spark process on an old table with avro encoding?